setup.py

```diff
@@ -1,30 +1,151 @@
-# README file and 2) it's easier to type in the README file than to put a raw
-# string in below ...
-def read(fname):
-    return open(os.path.join(os.path.dirname(__file__), fname)).read()
+
+PACKAGE_NAME = "libzim"
+VERSION = "6.1.1"  # pegged to be the same version as libzim since they are always released together
+LICENSE = "GPLv3"
+DESCRIPTION = "The official Python API for creating and interacting with ZIM files using libzim"
+AUTHOR = "Monadical Inc."
+AUTHOR_EMAIL = "jdc@monadical.com"
+GITHUB_URL = "https://github.com/openzim/python-libzim"
+
+BASE_DIR = Path(__file__).parent
+LIBZIM_CYTHON_DIR = 'libzim'
+
+INCLUDE_DIRS = [LIBZIM_CYTHON_DIR, 'include']
+LIBRARY_DIRS = ['lib']
+
+# use this option if you wish to manually specify the path to a prebuilt libzim release (.so and .h files)
+# set it to the path of the unzipped libzim subfolder containing zim/*.h
+LIBZIM_INCLUDE_DIR = os.getenv('LIBZIM_INCLUDE_DIR')
+if LIBZIM_INCLUDE_DIR:
+    if (Path(LIBZIM_INCLUDE_DIR) / 'zim/zim.h').exists():
+        INCLUDE_DIRS.insert(0, str(LIBZIM_INCLUDE_DIR))
+    else:
+        raise Exception(
+            f'Could not find zim/*.h files in LIBZIM_INCLUDE_DIR={LIBZIM_INCLUDE_DIR}\n'
+            f'    Hint: git clone --depth 1 -b tags/{VERSION} https://github.com/openzim/libzim libzim_cpp'
+            '          export LIBZIM_INCLUDE_DIR=$PWD/libzim_cpp/include'
+        )
+
+
+# use this option to manually specify the libzim dynamic library (aka "shared object") dir
+# set it to the path of the unzipped libzim subfolder containing libzim.so
+LIBZIM_LIBRARY_DIR = os.getenv('LIBZIM_LIBRARY_DIR')
+if LIBZIM_LIBRARY_DIR:
+    if not ((Path(LIBZIM_LIBRARY_DIR) / 'libzim.so').exists() or (Path(LIBZIM_LIBRARY_DIR) / 'libzim.a').exists()):
+        potential_dylibs = [f for f in Path(LIBZIM_LIBRARY_DIR).iterdir() if 'libzim.so' in str(f)]
+        raise Exception(
+            f'Could not find libzim.so file in LIBZIM_LIBRARY_DIR={LIBZIM_LIBRARY_DIR}\n' +
+            (
+                f'    Hint: ln -s {potential_dylibs[0].name} {LIBZIM_LIBRARY_DIR}/libzim.so'
+                if potential_dylibs else
+                f'    Hint: LIBZIM_LIBRARY_DIR should look something like .../libzim_linux-x86_64-{VERSION}/lib/x86_64-linux-gnu'
+            )
+        )
+    LIBRARY_DIRS.insert(0, str(LIBZIM_LIBRARY_DIR))
+
+# the default is to dynamically link (finds externally installed libzim on user's system)
+if not (LIBZIM_LIBRARY_DIR or find_library('zim')):
+    print(
+        '[!] Warning: Could not find libzim.so in available system libraries or LIBZIM_LIBRARY_DIR\n'
+        '    Hint: Install it from source from https://github.com/openzim/libzim\n'
+        '          or download a prebuilt zim release and set the env varaibles to point to it:\n'
+        f'    LIBZIM_INCLUDE_DIR=/libzim_linux-x86_64-{VERSION}/include\n'
+        f'    LIBZIM_LIBRARY_DIR=/libzim_linux-x86_64-{VERSION}/lib/x86_64-linux-gnu'
+    )
+
```

Comment:
All this thing about the diretories to find the libzim seems a bit to complex.
The user can simply set `CPPFLAGS=-I/.../include -L/.../lib` before running `setup.py` and everything is working fine.

To launch the wrapper, the user has to set `LD_LIBRARY_FLAGS=.../lib:$LD_LIBRARY_FLAGS`.

And so, there is nothing to do on setup.py to handle where the library is.

Reply:
`CPPFLAGS` is pretty foreign to the average python user, I was trying to get it so that even `LIBZIM_LIBRARY_DIR` doesn't need to be set most of the time.

In the interest of time though I can remove some of this autodetection in favor of requiring the user use more env vars.

Reply:
This is done ee61d2a